public/js/graph/react_graph.js

```diff
@@ -150,10 +150,21 @@ var Graph = React.createClass({
             zoomFactor: 1,
             horizontalPanFactor: 0,
             verticalPanFactor: 0,
-            mouseDown: false
+            mouseDown: false,
+            graphMode: 'graph',
+            drawNodeID: 0
         };
     },
 
+    componentWillMount: function () {
+        if (location.href.endsWith('draw')) {
+            this.setState({graphMode: 'draw-node'});
+            document.body.addEventListener('mousedown', this.drawGraphObject, false);
```

Comment:
You don't want to set the callback here; instead, it should be directly tied to the `svg` element that's rendered for this component.